### PR TITLE
Deploy fleet-manager from template in e2e tests

### PR DESCRIPTION
**Pull request by kovayur**
## Description
A big cleanup, the main goal of which is to **use the service template in E2E tests the same way we use it in app-interface**.

As a result:
1. Use fleet manager and postgres templates in E2E tests incl. CI. 
2. Deleted `fleet-manager` and `db` in `dev/env/manifests` that duplicate templates content.
3. Reduced the number of cluster configurations used in dev to one. Add an ability to override `cluster_dns` for openshift clusters in `Makefile` 
4. Use "no-tls" envoy configuration for non-openshift test clusters.
5. Cleanup the list of environment variables in dev scripts
6. Removed ginkgo from `tools`

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [x] Added test description under `Test manual`
- [x] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources

## Test manual
Tested on colima:
1. `make deploy/bootstrap`
2. `make deploy/dev`
3. ./.openshift-ci/tests/e2e.sh

### CODE REVIEW

File: .openshift-ci/tests/e2e.sh [hunk 1]
```diff
@@ -201,6 +199,6 @@ port-forwarding stop db 5432 || true
 port-forwarding stop fleet-manager 8000 || true
 
 log "Killing oc processes forcefully"
-pkill -9 oc || true
+pkill -9 '^oc$' || true
```
Comment (kovayur) [inline .openshift-ci/tests/e2e.sh:202]: This expects regexp and kills all processes with the substring "oc" in the name. It drives my laptop crazy

File: Makefile [hunk 2]
```diff
@@ -679,15 +683,14 @@ deploy/project:
 
 # deploy the postgres database required by the service to an OpenShift cluster
 deploy/db:
-	oc process -f ./templates/db-template.yml | oc apply -f - -n $(NAMESPACE)
+	oc process -f ./templates/db-template.yml --local | oc apply -f - -n $(NAMESPACE)
```
Comment (kovayur) [inline Makefile:686]: Add `--local` here and below to be able to render templates against *KS clusters

File: dev/config/dataplane-cluster-configuration.yaml [hunk 3]
```diff
@@ -18,14 +18,14 @@
 #    cluster_dns: apps.example.com #Valid cluster DNS. This will be used to build central host url and to communicate with standalone clusters. Required when "provider_type" is "standalone"
 #    supported_instance_type: "eval" # could be "eval", "standard" or both i.e "standard,eval" or "eval,standard". Defaults to "standard,eval" if not set
 clusters:
- - name: rancher-desktop
+ - name: dev
```
Comment (kovayur) [inline dev/config/dataplane-cluster-configuration.yaml:21]: The requirement for the cluster name to match the KUBECONFIG context name no longer exists, so we can squash these config files.

File: dev/config/gitops-config.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
+rhacsOperators:
+  crdUrls:
+    - https://raw.githubusercontent.com/stackrox/stackrox/4.2.1/operator/bundle/manifests/platform.stackrox.io_securedclusters.yaml
+    - https://raw.githubusercontent.com/stackrox/stackrox/4.2.1/operator/bundle/manifests/platform.stackrox.io_centrals.yaml
+  operators:
+    - deploymentName: "rhacs-operator-4.2.0-428-g318762a66d"
+      image: "quay.io/rhacs-eng/stackrox-operator:4.2.0-428-g318762a66d"  # should be in sync with operatorVersion1 defined in e2e_canary_upgrade_test.go
```
Comment (kovayur) [inline dev/config/gitops-config.yaml:7]: I guess there's a bug in canary upgrade test when the config is restored and we can't use an arbitrary version here.

File: dev/env/scripts/bootstrap.sh [hunk 5]
```diff
@@ -64,10 +55,8 @@ EOF
         fi
     fi
 
-    if [[ "$CLUSTER_TYPE"  == "kind" ]]; then
-        log "Ensuring operator images exist from dev GitOps config"
-        ensure_operator_image_exists.sh
-    fi
+    log "Ensuring operator images exist from dev GitOps config"
+    ensure_operator_image_exists.sh
```
Comment (kovayur) [inline dev/env/scripts/bootstrap.sh:59]: This fixes the problem on my colima cluster when `ose-kube-rbac-proxy`image cannot be pulled.

File: templates/envoy-config-configmap.yml
Comment (kovayur) [inline templates/envoy-config-configmap.yml:1]: This configuration is only used on dev.

File: templates/service-template.yml [hunk 6]
```diff
@@ -1163,7 +1168,7 @@ objects:
               optional: true
           initContainers:
           - name: migration
-            image: ${IMAGE_REGISTRY}/${IMAGE_REPOSITORY}:${IMAGE_TAG}
+            image: ${REPO_DIGEST}
```
Comment (kovayur) [inline templates/service-template.yml:1171]: Openshift CI has the full image name. `REPO_DIGEST` is also supported by app-interface as sha256 image digest. I left `IMAGE_TAG` as label for tracing purposes.

File: tools/tools.go [hunk 7]
```diff
@@ -11,7 +11,6 @@ import (
 	// Tool dependencies, not used anywhere in the code.
 	_ "github.com/go-bindata/go-bindata/v3"
 	_ "github.com/matryer/moq"
-	_ "github.com/onsi/ginkgo/v2/ginkgo"
```
Comment (kovayur) [inline tools/tools.go:14]: We often have a version missmatch between `go.mod` and `tools/go.mod`

**Review by ludydoo — APPROVED** [review]

**Pull request merged**